### ISSUE TITLE
[BUG] Headers designated as "required: false" but still expected in the response

### ISSUE THREAD

**Issue by hoog1511** [schemathesis/schemathesis#1629]
**Describe the bug**
I have an openAPI schema where there are optional headers for a 404 response. This is due to a complex configuration where multiple connected services and a gateway can send the response with different contentTypes for the API I am testing. Depending on the contentType the headers are present i.e., application/json has them and text/html does not. During the schemathesis test with CLI the results of the test show that the response headers are required even though I designated them as optional either by ommitting the header being required or by designating it as "required: false".

To my knowledge there is no way to designate separate responses with headers in openAPI specifiction. If there is a way to do this could you, please inform me on a possible solution.
`

**Expected behavior**
A passing schemathesis test with a schema which has optional headers.

**Environment (please complete the following information):**

- OS: Windows
- Python version: latest
- Schemathesis version: latest
- Spec version: 3.0.3

**Comment by Stranger6667** [comment]
Hi @hoog1511 

Thank you for reporting! That is strange as the exact issue should have been fixed as a part of the `3.16.0` release via https://github.com/schemathesis/schemathesis/pull/1564. Am I missing something?

Could you, please take a look if we talk about the same aspect of the validation process?

**Comment by hoog1511** [comment]
hi @Stranger6667 

I do seem to be using an outdated version of schemathesis. I tried updating it but now I keep receiving error messages regarding a missing httpx and exceptiongroup package. I added these manually, but the tests are not running for me and keep returning an error. 
command:
`schemathesis run static/yaml.yaml -c status_code_conformance -c content_type_conformance -c response_headers_conformance -c response_schema_conformance --base-url https://someurl.com/ --auth "..." --show-errors-tracebacks --request-tls-verify=false --store-network-log="TRACEBACK.yaml" --debug-output-file="outputFile.yaml" --hypothesis-deadline=None --hypothesis-suppress-health-check=too_slow,filter_too_much --hypothesis-max-examples=50 --junit-xml=unit.xml`

error (see image)
<img width="1175" alt="error" src="https://user-images.githubusercontent.com/22965220/200333134-9924185d-2f20-4748-b47d-6381f22fc9b6.png">
